assets/app/view/game/buy_trains.rb

```diff
@@ -86,10 +86,12 @@ def render
 
         if (round.can_buy_train? && round.corp_has_room?) || round.must_buy_train?
           children << h(:h3, h3_props, 'Available Trains')
-          children << h(:div, div_props, [
-            *from_depot(depot_trains),
-            *other_corp_trains.any? ? other_trains(other_corp_trains) : '',
-          ])
+          lines = [*from_depot(depot_trains)]
+          @corporation.abilities(:train_discount) do |ability|
+            lines += [*from_depot_discounted(depot_trains, ability)]
```

Comment:
redundant

and better to do lines.concat(from_depot)

Reply:
I refactored this so that the "gritty stuff" is done in from_depot.